vimeo-networking/src/main/java/com/vimeo/networking/model/VideoFile.java

```diff
@@ -24,21 +24,22 @@
 
 import com.google.gson.annotations.SerializedName;
 import com.vimeo.stag.UseStag;
+import com.vimeo.stag.UseStag.FieldOption;
 
+import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
 import java.io.Serializable;
 import java.util.Date;
 
 /**
- * Representation of a Video stream/playback file
+ * Base representation of a Video stream/playback file
  * <p>
  * Created by alfredhanssen on 4/25/15.
  */
 @SuppressWarnings("unused")
-// TODO: Remove the VideoFileDeserializer and use Stag instead, once API corrects log issue behind version 2/16/17 [KZ]
-// @UseStag(FieldOption.SERIALIZED_NAME)
-public class VideoFile implements Serializable {
+@UseStag(FieldOption.SERIALIZED_NAME)
+public abstract class VideoFile implements Serializable {
```

Comment:
I can see how the inheritance hierarchy makes sense in the way that @kevinzetterstrom refactored it, especially since the server is providing the different objects as such. However, the casting of the objects that is happening at a higher level (in VimeoPlayer) suggests that the design of the object consumption on the app level is not ideal. We should try to address that issue in the related vimeo-android ticket.

One related issue here is that the MimeType should not be in the VideoFile class. It is only relevant to the ProgressiveVideoFileClass.